src/lwc/geFormWidgetRowAllocation/geFormWidgetRowAllocation.js

```diff
@@ -88,153 +58,125 @@ export default class GeFormWidgetRowAllocation extends LightningElement {
+        // if(event.detail.targetFieldName === ALLOCATION_PERCENT) {
+        //     // payload = this.handlePercentChange(event.detail, totalForCalculation);
+        // }
+        // if(isNotEmpty(payload)) {
+        //     // notify listeners of the new allocation values
+        //     fireEvent(null, 'allocationValueChange', {rowIndex: this.rowIndex, payload});
+        // }
+    }
+
+    disablePercentFieldIfAmountHasValue(changedField) {
+        let percentFieldElement = this.template.querySelector(
+    `[data-fieldname=${this.allocationPercentageFieldApiName}]`
+        );
+        if(isNumeric(changedField.value) && changedField.value > 0 && !percentFieldElement.disabled) {
+            percentFieldElement.disabled = true;
+        } else if(isEmpty(changedField.value) || changedField.value === 0) {
+            percentFieldElement.disabled = false;
+        }
     }
 
-    /**
-     * Handle field updates inside this allocation row.
-     * @param event
-     * @param total Optional total amount parameter for when we're reacting to the total donation amount changing
-     */
-    handleChange(event, total) {
-        let payload;
-        // if a total was passed in, use it for calculating percent allocations, otherwise use this.totalAmount
-        const totalForCalculation = isNumeric(total) ? total : this.totalAmount;
-        if(event.detail.targetFieldName === ALLOCATION_PERCENT) {
-            // handle the percent field being updated
-            payload = this.handlePercentChange(event.detail, totalForCalculation);
-        } else if(event.detail.targetFieldName === ALLOCATION_AMOUNT) {
-            // handle the Amount field being updated
-            payload = this.handleAmountChange(event.detail);
+    disableAmountFieldIfPercentHasValue(changedField) {
+        let amountFieldElement = this.template.querySelector(
+            `[data-fieldname=${this.allocationAmountFieldApiName}]`
+        );
+        if(isNumeric(changedField.value) && changedField.value > 0 && !amountFieldElement.disabled) {
+            amountFieldElement.disabled = true;
+        } else if(isEmpty(changedField.value) || changedField.value === 0) {
+            amountFieldElement.disabled = false;
         }
+    }
 
-        if(isNotEmpty(payload)) {
-            // notify listeners of the new allocation values
-            fireEvent(null, 'allocationValueChange', {rowIndex: this.rowIndex, payload});
+    calculateAmountFromPercent(percentValue) {
+        const percentDecimal = parseFloat(percentValue) / 100;
+        // convert to cents instead of dollars, avoids floating point problems
+        const totalInCents = this.widgetDataFromState[this.donationAmountFieldApiName] * 100;
+
+        return {
+            [this.allocationAmountFieldApiName] : Math.round(totalInCents * percentDecimal) / 100
         }
     }
 
-    getFieldByName(fieldName) {
-        return this.template.querySelector(`[data-fieldname='${fieldName}']`);
+    remove() {
```

Comment:
@dfuller114  Can we change the name of this function to be more descriptive? maybe `removeRow` or `deleteRow` ?